cores/esp8266/Tone.cpp

```diff
@@ -49,7 +52,7 @@ void tone(uint8_t _pin, unsigned int frequency, unsigned long duration) {
   if (frequency == 0) {
     noTone(_pin);
   } else {
-    uint32_t period = (1000000L * system_get_cpu_freq()) / frequency;
+    uint32_t period = microsecondsToClockCycles(1000000UL) / frequency;
```

Comment:
I'm not sure whether this change is ok. The microsecondsToClockCycles() define uses the F_CPU macro, while system_get_cpu_freq() is the SDK api call. 
I _think_ it is ok because these functions aren't supposed to be called from an ISR. Also, if built for 160MHz, execution will be done at that speed something like 95% of the time.